src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -116,73 +134,116 @@ public ActivityTraceFlags TraceFlags
+        /// cref="OpenTelemetryLoggerOptions.IncludeTraceState"/> is enabled.
+        /// </remarks>
         public string? TraceState
         {
-            get => this.Data.TraceState;
-            set => this.Data.TraceState = value;
+            get => this.ILoggerData.TraceState;
+            set => this.ILoggerData.TraceState = value;
         }
 
         /// <summary>
         /// Gets or sets the log category name.
         /// </summary>
         public string? CategoryName
         {
-            get => this.Data.CategoryName;
-            set => this.Data.CategoryName = value;
+            get => this.ILoggerData.CategoryName;
+            set => this.ILoggerData.CategoryName = value;
         }
 
         /// <summary>
         /// Gets or sets the log <see cref="Microsoft.Extensions.Logging.LogLevel"/>.
         /// </summary>
         public LogLevel LogLevel
         {
-            get => this.Data.LogLevel;
-            set => this.Data.LogLevel = value;
+            get => this.ILoggerData.LogLevel;
+            set => this.ILoggerData.LogLevel = value;
         }
 
         /// <summary>
         /// Gets or sets the log <see cref="Microsoft.Extensions.Logging.EventId"/>.
         /// </summary>
         public EventId EventId
         {
-            get => this.Data.EventId;
-            set => this.Data.EventId = value;
+            get => this.ILoggerData.EventId;
+            set => this.ILoggerData.EventId = value;
         }
 
         /// <summary>
         /// Gets or sets the log formatted message.
         /// </summary>
+        /// <remarks>
+        /// Note: Only set if <see
+        /// cref="OpenTelemetryLoggerOptions.IncludeFormattedMessage"/> is
+        /// enabled.
+        /// </remarks>
         public string? FormattedMessage
         {
-            get => this.Data.Message;
-            set => this.Data.Message = value;
+            get => this.ILoggerData.FormattedMessage;
+            set => this.ILoggerData.FormattedMessage = value;
         }
 
         /// <summary>
-        /// Gets or sets the raw state attached to the log. Set to <see
-        /// langword="null"/> when <see
+        /// Gets or sets the log body.
+        /// </summary>
+        /// <remarks>
+        /// Note: Set to the <c>{OriginalFormat}</c> attribute (message
+        /// template) if found otherwise the formatted log message.
+        /// </remarks>
+        public string? Body
+        {
+            get => this.Data.Body;
+            set => this.Data.Body = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the raw state attached to the log.
+        /// </summary>
+        /// <remarks>
+        /// Note: Set to <see langword="null"/> when <see
         /// cref="OpenTelemetryLoggerOptions.ParseStateValues"/> is enabled.
+        /// </remarks>
+        [Obsolete("State cannot be accessed safely outside of an ILogger.Log call stack. It will be removed in a future version.")]
```

Comment:
Should we add any more detail to this obsolete message? Basically, what should one do if they see this obsolete message.

Reply:
If I'm the user, I most likely will suppress the warnings in my build until there is a non-backward compatible version got launched and I have to update to that version.

Reply:
> Should we add any more detail to this obsolete message? Basically, what should one do if they see this obsolete message.

I updated the message so it has more guidance.

> If I'm the user, I most likely will suppress the warnings in my build until there is a non-backward compatible version got launched and I have to update to that version.

Fine if that's what people want to do. We may never actually remove the obsolete things 😄 The audience for `LogRecord` is primarily exporter authors. The number of exporters out there really shouldn't be huge so hopefully we can assist with these updates when they are available.